mypy/report.py

```diff
@@ -813,7 +821,13 @@ def on_file(self,
                 modules: Dict[str, MypyFile],
                 type_map: Dict[Expression, Type],
                 options: Options) -> None:
-        path = os.path.relpath(tree.path)
+
+        try:
+            path = os.path.relpath(tree.path)
+
+        except ValueError:
+            path = tree.path
+
```

Comment:
What is the semantic reason for making the path relative? When I understand the code correctly a relative path outside of the directory where the code-under-check is in, should be ignored anyway right (see `should_skip_path` case `...`)?
So a final solution could be to simple return from this function in case of a ValueError? Or to mark the resulting path as __invalid__ and skip it with `should_skip_path`. What do you think?

Reply:
We use relative paths since the paths could be included in the output and we want the output to be tidy and not depend on the current working directory. I agree that if we can't determine a relative path, we can just skip and return (both here and above).

Reply:
I solved it by returning directly from the exception handler. Another alternative would be to mark the path as invalid (None or empty string for example) and skip then from the function `should_skip_path(...)`. However this would introduce special state for path, which is not obvious, especially in cases where the function `should_skip_path(...)` is not used (in future). On the other hand with the current solution there are two logics to skip: the official one from `should_skip_path(...)` and the new one from the exception handler. 
Just decide by yourself, which solution you like more. I can change it another time, if you want.